utils/writeable/src/impls.rs

```diff
@@ -29,7 +29,7 @@ macro_rules! impl_write_num {
             }
 
             fn writeable_length_hint(&self) -> $crate::LengthHint {
-                LengthHint::exact(self.checked_ilog10().unwrap_or(0) as usize + 1)
+                LengthHint::exact(impls::ILog10Ext::checked_ilog10(*self).unwrap_or(0) as usize + 1)
```

Comment:
This was shadowing on purpose! We want to use the inherent method if it's available, and fall back to the trait. #3076

Reply:
oh, good catch

Reply:
good catch! fixed!